apps/src/JSInterpreter.js

```diff
@@ -104,6 +105,9 @@ var JSInterpreter = module.exports = function (options) {
   };
 
   try {
+    // Return value will be stored as this.interpreter inside the supplied
+    // initFunc() (other code in initFunc() depends on this.interpreter, so
+    // we can't wait until the constructor returns)
```

Comment:
This strikes me as a code smell then.  Doesn't this essentially indicate a circular dependency?  It sounds like the `Interpreter` being constructed depends on the thing constructing it (the `JSInterpreter`) which in turn needs the `Interpreter` to be fully constructed first.  Assigning `this.interpreter` halfway through construction (giving the half-constructed member access to its half-constructed parent) feels like a band-aid.  I'm afraid it will be painful to work around when this initialization gets even more complex.

I'm not sure what the solution is yet, though, and I know this isn't new code.  Don't let it block this PR, but let's see if we can solve this cleanly.